frontend/test/pytest/test_cuda_integration.py

```diff
@@ -0,0 +1,459 @@
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""CUDA Integration testing."""
+
+import jax
+import pennylane as qml
+import pytest
+from jax import numpy as jnp
+from numpy.testing import assert_allclose
+
+from catalyst import measure, qjit
+from catalyst.compilation_pipelines import QJIT
+from catalyst.compiler import CompileOptions
+from catalyst.utils.jax_extras import remove_host_context
+
+# This import is here on purpose. We shouldn't ever import CUDA
+# when we are running kokkos. Importing CUDA before running any kokkos
+# kernel polutes the environment and will create a segfault.
+# pylint: disable=import-outside-toplevel
+
+
+@pytest.mark.cuda
+class TestCuda:
+    """CUDA integration tests. Skip if kokkos."""
+
+    def test_argument(self):
+        """Test that we can pass cuda_quantum as a compiler to @qjit decorator."""
+
+        @qjit(compiler="cuda_quantum")
+        @qml.qnode(qml.device("lightning.qubit", wires=1))
+        def circuit_foo():
+            return qml.state()
+
+    def test_qjit_cuda_generate_jaxpr(self):
+        """Test the JAXPR generation."""
+
+        from catalyst.cuda.catalyst_to_cuda_interpreter import QJIT_CUDA
+
+        @qml.qnode(qml.device("lightning.qubit", wires=1))
+        def circuit_foo():
+            return qml.state()
+
+        opts = CompileOptions()
+        expected_jaxpr = QJIT(circuit_foo, opts).jaxpr
+        observed_jaxpr, _ = QJIT_CUDA(circuit_foo).get_jaxpr()
+        assert str(expected_jaxpr) == str(observed_jaxpr)
+
+    def test_qjit_cuda_remove_host_context(self):
+        """Test removing the host context."""
+
+        from catalyst.cuda.catalyst_to_cuda_interpreter import QJIT_CUDA
+
+        @qml.qnode(qml.device("lightning.qubit", wires=1))
+        def circuit_foo():
+            return qml.state()
+
+        observed_jaxpr, _ = QJIT_CUDA(circuit_foo).get_jaxpr()
+        jaxpr = remove_host_context(observed_jaxpr)
+        assert jaxpr
+
+    def test_qjit_catalyst_to_cuda_jaxpr(self):
+        """Assert that catalyst_to_cuda returns something."""
+        from catalyst.cuda.catalyst_to_cuda_interpreter import interpret
+
+        @qml.qnode(qml.device("lightning.qubit", wires=1))
+        def circuit_foo():
+            return qml.state()
+
+        cuda_jaxpr = jax.make_jaxpr(interpret(circuit_foo))
+        assert cuda_jaxpr
+
+    def test_qjit_catalyst_to_cuda_jaxpr_actually_call_ry(self):
+        """Assert that catalyst_to_cuda matches the expected results."""
+        import cudaq
+
+        from catalyst.cuda.catalyst_to_cuda_interpreter import interpret
```

Comment:
would you like to add this to the header?

Reply:
No. Because if we import `cudaq` it breaks. Even if we skip the tests in this file, the `import`s happen. So all kokkos tests would fail.